appIcons.js

```diff
@@ -187,6 +191,13 @@ const DockAbstractAppIcon = GObject.registerClass({
             }
         });
 
+        this.connect('notify::updating', () => {
+            const icon = this.icon._iconBin;
+            if (this.updating)
+                icon.set_opacity(128);
+            else
+                icon.set_opacity(255);
```

Comment:
I'm wondering if these would be better as `set_opacity_override`, so as to prevent losing any previous custom opacity value. But I won't block on that -- it would only hurt the dock itself if such a loss ever happens so we'll deal with that later.